migrations/2024-08-27-153454_create-challenges-table/up.sql

```diff
@@ -0,0 +1,11 @@
+-- Your SQL goes here
+CREATE TABLE challenges (
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    title VARCHAR(255) NOT NULL,
+    description TEXT NOT NULL,
+    repo_url VARCHAR(255) NOT NULL,
+    difficulty VARCHAR(255) CHECK (difficulty IN ('easy', 'medium', 'hard')) NOT NULL,
+    mode VARCHAR(255) CHECK (mode IN ('project', 'functional_test')) NOT NULL,
+    created_at TIMESTAMP NOT NULL DEFAULT now(),
+    updated_at TIMESTAMP NOT NULL DEFAULT now()
```

Comment:
This will be set to datetime at time of row creation and will need to be updated manually on every update after that (unless we add a database trigger).  Just something to note.